pastebin-workitem/src/main/java/org/jbpm/process/workitem/pastebin/GetExistingPastebinWorkitemHandler.java

```diff
@@ -0,0 +1,97 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jbpm.process.workitem.pastebin;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.jbpm.process.workitem.core.AbstractLogOrThrowWorkItemHandler;
+import org.jbpm.process.workitem.core.util.RequiredParameterValidator;
+import org.jbpm.process.workitem.core.util.Wid;
+import org.jbpm.process.workitem.core.util.WidMavenDepends;
+import org.jbpm.process.workitem.core.util.WidParameter;
+import org.jbpm.process.workitem.core.util.WidResult;
+import org.jbpm.process.workitem.core.util.service.WidAction;
+import org.jbpm.process.workitem.core.util.service.WidService;
+import org.jpastebin.pastebin.Pastebin;
+import org.jpastebin.pastebin.PastebinLink;
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Wid(widfile = "GetPastebin.wid", name = "GetPastebin",
+        displayName = "GetPastebin",
+        defaultHandler = "mvel: new org.jbpm.process.workitem.pastebin.GetExistingPastebinWorkitemHandler()",
+        documentation = "${artifactId}/index.html",
+        parameters = {
+                @WidParameter(name = "PastebinKey", required = true)
+        },
+        results = {
+                @WidResult(name = "PasteContent")
+        },
+        mavenDepends = {
+                @WidMavenDepends(group = "${groupId}", artifact = "${artifactId}", version = "${version}")
+        },
+        serviceInfo = @WidService(category = "${name}", description = "${description}",
+                keywords = "paste,pastebin,get,existing",
+                action = @WidAction(title = "Get an existing Pastebin")
+        ))
+public class GetExistingPastebinWorkitemHandler extends AbstractLogOrThrowWorkItemHandler {
+
+    private String develKey;
+
+    private static final Logger logger = LoggerFactory
+            .getLogger(CreatePastebinWorkitemHandler.class);
+    private static final String RESULTS_VALUE = "PasteContent";
+
+    public GetExistingPastebinWorkitemHandler(String develKey) {
+        this.develKey = develKey;
+    }
+
+    public void executeWorkItem(WorkItem workItem,
+                                WorkItemManager workItemManager) {
+        try {
+            RequiredParameterValidator.validate(this.getClass(),
+                                                workItem);
+
+            String pastebinKey = (String) workItem.getParameter("PastebinKey");
+            // for testing
+            PastebinLink testPastebinLink = (PastebinLink) workItem.getParameter("TestPastebinLink");
```

Comment:
similar here, allow to provide PastebinLink as work item param so it won't be test specific

Reply:
done here too. thanks.